source/System.cpp

```diff
@@ -42,6 +42,23 @@ namespace {
 	const double VOLUME = 2000.;
 	// Above this supply amount, price differences taper off:
 	const double LIMIT = 20000.;
+
+	template<class Container>
+	void LoadRandomIntervalConditions(Container &container, const DataNode &child)
+	{
+		if(container.empty())
+			return;
+		shared_ptr<ConditionSet> conditions = container.back().GetConditions();
+		for(auto & grand : child)
+			if(grand.Size() != 2 || grand.Token(0) != "to" || grand.Token(1) != "spawn" || !grand.HasChildren())
+				continue;
+			else if(!conditions)
```

Comment:
```suggestion
			if(!conditions)
```

Reply:
That would put it outside the `for` loop.

Reply:
That "if" statement is gone and the function is cleaner now.